components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/SubscriptionsApiServiceImpl.java

```diff
@@ -1,42 +1,115 @@
 import org.wso2.carbon.apimgt.rest.api.dto.SubscriptionDTO;
+import org.wso2.carbon.apimgt.rest.api.exception.InternalServerErrorException;
+import org.wso2.carbon.apimgt.rest.api.exception.NotFoundException;
+import org.wso2.carbon.apimgt.rest.api.utils.RestApiUtil;
+import org.wso2.carbon.apimgt.rest.api.utils.mappings.APIMappingUtil;
+import org.wso2.carbon.apimgt.rest.api.utils.mappings.SubscriptionMappingUtil;
 
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
-
-import java.io.InputStream;
-
+import java.util.Set;
 import javax.ws.rs.core.Response;
 
 public class SubscriptionsApiServiceImpl extends SubscriptionsApiService {
     @Override
-    public Response subscriptionsGet(String apiId,String applicationId,String accept,String ifNoneMatch){
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
-    }
-    @Override
-    public Response subscriptionsPost(SubscriptionDTO body,String contentType){
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+    public Response subscriptionsGet(String apiId, String applicationId, String groupId, String accept,
+            String ifNoneMatch) {
+
+        String username = RestApiUtil.getLoggedInUsername();
+        Subscriber subscriber = new Subscriber(username);
+        Set<SubscribedAPI> subscriptions = new HashSet<>();
+        try {
+            APIConsumer apiConsumer = RestApiUtil.getConsumer(username);
+            if (!StringUtils.isEmpty(apiId)) {
+                APIIdentifier apiIdentifier = APIMappingUtil.getAPIIdentifier(apiId);
+                subscriptions = apiConsumer.getSubscribedIdentifiers(subscriber, apiIdentifier, groupId);
+
+            } else if (!StringUtils.isEmpty(applicationId)) {
+                Application application = apiConsumer.getApplicationById(applicationId);
+                subscriptions =
+                        apiConsumer.getSubscribedAPIs(subscriber, application.getName(), application.getGroupId());
+            }
+
+            List<SubscriptionDTO> subscriptionDTOs = new ArrayList<>();
+            for (SubscribedAPI subscription : subscriptions) {
+                SubscriptionDTO subscriptionDTO = SubscriptionMappingUtil.fromSubscriptiontoDTO(subscription);
+                //when retrieving subscriptions from getSubscribedAPIs(), existing application ids are not populated
+                if (!StringUtils.isEmpty(applicationId)) {
+                    subscriptionDTO.setApplicationId(applicationId);
+                }
+                subscriptionDTOs.add(subscriptionDTO);
+            }
+            return Response.ok().entity(subscriptionDTOs).build();
+
+        } catch (APIManagementException e) {
+            throw new InternalServerErrorException(e);
```

Comment:
Fixed in #1303 (https://github.com/wso2/carbon-apimgt/commit/8049b7e2779abb3dd6bf13aeb70b062438e7c5b7) and #1533 (https://github.com/wso2/carbon-apimgt/commit/243966339e63e60a915cb07ee4e5562a9cb7facf)